askama_shared/src/input.rs

```diff
@@ -129,6 +131,34 @@ impl<'a> TemplateMeta<'a> {
     }
 }
 
+pub struct Imports<'a> {
+    pub sources: Vec<Cow<'a, str>>
+}
+
+impl <'a> Imports<'a> {
+    pub fn new(parent_nodes: &'a [Node], parent_path: &'a Path) -> Imports<'a> {
```

Comment:
Upon reflection, I guess the template sources have to be preserved, since the AST nodes depend on them. Still, I think we can extract the `Macro` nodes earlier?

Reply:
The first step is needed for ownership reasons but I think the other two can be combined. Then Imports::parse() would return the macros directly.